include/boost/json/impl/pointer.ipp

```diff
@@ -0,0 +1,329 @@
+    pointer_token(
+        string_view::const_iterator b,
+        string_view::const_iterator e) noexcept
+        : b_(b), e_(e)
+    {
+    }
+
+    iterator begin() const noexcept;
+    iterator end() const noexcept;
+
+private:
+    string_view::const_iterator b_;
+    string_view::const_iterator e_;
+};
+
+class pointer_token::iterator
+{
+public:
+    explicit iterator(string_view::const_iterator base) noexcept
+        : base_(base)
+    {
+    }
+
+    char operator*() const noexcept
+    {
+        switch (char c = *base_)
+        {
+        case '~':
+            c = base_[1];
+            if ( '0' == c )
+            {
+                return '~';
+            }
+            else
+            {
+                BOOST_ASSERT('1' == c);
+                return '/';
+            }
+        default:
+            return c;
+        }
+    }
+
+    iterator& operator++() noexcept
+    {
+        if ( '~' == *base_ )
+        {
+            base_ += 2;
+        }
+        else
+        {
+            ++base_;
+        }
+        return *this;
+    }
+
+    string_view::const_iterator base() const noexcept
+    {
+        return base_;
+    }
+
+private:
+    string_view::const_iterator base_;
+};
+
+bool operator==(pointer_token::iterator l, pointer_token::iterator r) noexcept
+{
+    return l.base() == r.base();
+}
+
+bool operator!=(pointer_token::iterator l, pointer_token::iterator r) noexcept
+{
+    return l.base() != r.base();
+}
+
+pointer_token::iterator pointer_token::begin() const noexcept
+{
+    return iterator(b_);
+}
+
+pointer_token::iterator pointer_token::end() const noexcept
+{
+    return iterator(e_);
+}
+
+bool operator==(pointer_token token, string_view sv) noexcept
+{
+    auto t_b = token.begin();
+    auto const t_e = token.end();
+    auto s_b = sv.begin();
+    auto const s_e = sv.end();
+    for (; t_b != t_e && s_b != s_e; ++t_b, ++s_b)
+    {
+        if (*t_b != *s_b)
+        {
+            return false;
+        }
+    }
+
+    return t_e == t_b && s_e == s_b;
+}
+
+std::size_t
+parse_number_token(
+    string_view::const_iterator& b,
+    string_view::const_iterator e,
+    error_code& ec) noexcept
+{
+    if ( ( b == e ) || ( *b == '0' && ( b + 1 ) != e  && b[1] != '/' ) )
+    {
+        ec = error::token_not_number;
+        return {};
+    }
+
+    std::size_t result = 0;
+    for (; b != e; ++b)
+    {
+        char const c = *b;
+
+        if ( '/' == c)
+        {
+            break;
+        }
+
+        if ( c < '0' || c > '9' )
+        {
+            ec = error::token_not_number;
+            return {};
+        }
+
+        constexpr std::size_t max_size =
+            (std::numeric_limits<std::size_t>::max)();
+        constexpr std::size_t max_div10 = max_size / 10;
+
+        if ( result <= max_div10 )
+        {
+            result *= 10;
+            std::size_t d = c - '0';
+            if ( result <= max_size - d )
+            {
+                result += d;
+                continue;
```

Comment:
There's an easier way to check for unsigned overflow... just test if the new value is _smaller_ than the old value (i.e. it wrapped around).